shadow-dom/events/event-dispatch/test-003.html

```diff
@@ -29,31 +29,53 @@
     var d = newRenderedHTMLDocument(ctx);
 
     var host = d.createElement('div');
-    host.setAttribute('style', 'height:50%; width:100%');
     host.setAttribute('id', 'host');
     d.body.appendChild(host);
 
     //Shadow root to play with
     var s = host.createShadowRoot();
+    s.id = 'shadow';
 
-    var div1 = d.createElement('div');
-    div1.setAttribute('style', 'height:100%; width:100%');
-    div1.setAttribute('id', 'div1');
-    s.appendChild(div1);
-
-    s.addEventListener('mouseover', A_05_05_03_T01.step_func(function(event) {
-    	assert_true(false, 'Event listeners shouldn\'t be invoked if target and relatedTarget ' +
-    			'are the same');
-    }), false);
+    var input1 = d.createElement('input');
+    input1.setAttribute('id', 'input1');
+    s.appendChild(input1);
 
+    var input2 = d.createElement('input');
+    input2.setAttribute('id', 'input2');
+    s.appendChild(input2);
 
-    var evt = document.createEvent("MouseEvents");
-    evt.initMouseEvent("mouseover", true, false, window,
-      0, 10, 10, 10, 10, false, false, false, false, 0, div1);
+    // Add an event listener in its shadow root.
+    s.addEventListener('focusin', A_05_05_03_T01.step_func(function(event) {
```

Comment:
Why don't you add an event listener to a input1, or input2, to avoid `if elseif else`?